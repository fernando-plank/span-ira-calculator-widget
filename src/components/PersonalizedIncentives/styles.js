import styled from 'styled-components'

export const Wrapper = styled.div`
  width: fill-available;
  background-color: ${({ theme }) => theme.colors.primary};
  padding: 50px 24px;

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    margin: 0px 30px;
  }

  svg {
    margin: 0px 10px;
  }
`

export const Title = styled.h1`
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  text-align: left;
  margin-bottom: 42px;
  color: ${({ theme }) => theme.colors.white};

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    width: 30vw;
  }
`

export const IncentivesBody = styled.div`
  display: flex;
  margin-bottom: 50px;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    flex-direction: column-reverse;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    flex-direction: row;
  }
`

export const IncentivesBodyIcon = styled.div`
  width: 50%;
  display: flex;
  justify-content: flex-start;
  align-items: end;
  padding-left: 40px;
  height: 30vh;

  @media (max-width: ${({ theme }) => theme.breakpoints.mobileL}) {
    position: absolute;
    align-self: end;
    margin-top: 100px;
  }
`

export const DescriptionWrapper = styled.div`
  display: flex;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    flex-direction: column;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    flex-direction: row;
    width: 100%;
  }
`

export const DescriptionContainer = styled.div`
  color: ${({ theme }) => theme.colors.white};

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    width: 50%;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    margin-top: 28px;
  }
`

export const Subtitle = styled.h6`
  font-style: normal;
  font-weight: 500;
  font-size: 24px;
  line-height: 24px;
  text-transform: uppercase;
  text-align: left;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    max-width: 47vw;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    max-width: 90%;
  }

  svg {
    & > path {
        fill: white;
    }

    & > circle {
      stroke: white;
    }
  }
`

export const Price = styled.h6`
  font-size: 38px;
  font-weight: 700;
  line-height: 40px;
  letter-spacing: 0em;
  text-align: left;
`

export const IncentivesFooter = styled.div`
  display: flex;
  margin-top: 40px;
`

export const IncentivesFooterContent = styled.p`
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
  padding-right: 59px;
  text-align: left;
`

export const Separator = styled.div`
  border: 2px solid ${({ theme }) => theme.colors.white};
`;

export const TotalWrapper = styled.div`
  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    width: 80%;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    width: 100%;
  }
`;
