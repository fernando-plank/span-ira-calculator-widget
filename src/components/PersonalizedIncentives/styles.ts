import styled from 'styled-components'

export const Wrapper = styled.div`
  width: 100%;
  background-color: ${({ theme }) => theme.colors.white};

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    padding: 0px 41px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptopL}) {
    padding: 40px 0px;
  }

  @media (max-width: ${({ theme }) => theme.breakpoints.mobileL}) {
    .hide-on-mobile {
      display: none;
    }
  }
`

export const Background = styled.div`
  background-color: ${({ theme }) => theme.colors.primary};

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    padding: 50px 24px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileXL}) {
    padding: 50px 32px;
  }

  @media (min-width: ${(props) => props.theme.breakpoints.tablet}) {
    padding: 50px 40px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    padding: 64px 41px 40px 32px;
  }
`

export const Content = styled.div`
  @media (min-width: ${({ theme }) => theme.breakpoints.laptopL}) {
    padding: 0px;
    width: 1400px;
    margin: 0px auto;
  }

  svg {
    margin: 0px 10px;
  }
`

export const Title = styled.h1`
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  text-align: left;
  margin-bottom: 42px;
  color: ${({ theme }) => theme.colors.white};
  width: 280px;
  
  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    width: 380px;
  }
`

export const IncentivesBody = styled.div`
  display: flex;
  margin-bottom: 50px;

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    flex-direction: column-reverse;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    flex-direction: row;
    justify-content: space-between;
  }
`

export const IncentivesBodyIcon = styled.div`
  width: 50%;
  display: flex;
  justify-content: flex-start;
  align-items: end;

  @media (max-width: ${({ theme }) => theme.breakpoints.tablet}) {
    margin-top: 15px;
  }
`

export const DescriptionWrapper = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    //width: 100%;
  }
`

export const DescriptionBox = styled.div`
  display: flex;

  @media (max-width: ${({ theme }) => theme.breakpoints.tablet}) {
    flex-direction: column;
  }
`

export const DescriptionContainer = styled.div`
  color: ${({ theme }) => theme.colors.white};

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    margin-right: 50px;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    margin-top: 28px;
  }
`

export const TotalContainer = styled.div`
  margin-top: 44px;
  color: ${({ theme }) => theme.colors.white};
`

export const TotalText = styled.h1`
  font-size: 48px;
  line-height: 56px;
  font-weight: 700;
`

export const TotalPrice = styled.h1`
  font-size: 48px;
  line-height: 56px;
  font-weight: 700;

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    font-size: 136px;
    font-weight: 600;
    line-height: 136px;
  }
`

export const Subtitle = styled.h6`
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 20px;
  text-transform: uppercase;
  text-align: left;
  max-width: 100vw;

  svg {
    & > path {
      fill: white;
    }

    & > circle {
      stroke: white;
    }
  }
`

export const Price = styled.h6`
  font-size: 38px;
  font-weight: 700;
  line-height: 40px;
  letter-spacing: 0em;
  text-align: left;
`

export const IncentivesFooter = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 72px;
  
  > :last-child {
    margin-right: 31px;
  }

  @media (max-width: ${({ theme }) => theme.breakpoints.laptop}) {
    flex-direction: column;
    margin-top: 40px;
  }
`

export const IncentivesFooterContent = styled.p`
  color: ${({ theme }) => theme.colors.white} !important;
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
  text-align: left;
  width: 522px;

  margin-bottom: 30px;

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    //padding-right: 59px;
  }
`

export const ContentLink = styled.a`
  text-decoration: none;
  color: white;
  font-size: 12px;
  margin: 12px 0px 9px 0px;
  font-weight: bold;
`

export const Separator = styled.div`
  border: 1px solid ${({ theme }) => theme.colors.white};
`

export const TotalWrapper = styled.div`
  @media (min-width: ${({ theme }) => theme.breakpoints.mobileS}) {
    width: 80%;
  }

  @media (min-width: ${({ theme }) => theme.breakpoints.laptop}) {
    width: 100%;
  }
`
