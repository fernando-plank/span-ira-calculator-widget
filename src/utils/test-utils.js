var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
