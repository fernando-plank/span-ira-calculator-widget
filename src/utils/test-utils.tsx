/* eslint-disable import/export */
import { ReactElement } from 'react'
import { RenderOptions, render } from '@testing-library/react'

import theme from '@styles/theme'
import { ThemeProvider } from 'styled-components'

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (ui: ReactElement, { ...options }: RenderOptions = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options)

export * from '@testing-library/react'
export { customRender as render }
