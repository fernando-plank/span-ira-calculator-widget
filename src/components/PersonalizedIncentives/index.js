import React from 'react';
import Icon from '@components/Icon';
import Info from '@components/Info';
import { formatCurrency } from '@utils/currency';
import * as S from './styles';
const PersonalizedIncentives = ({ householdInformation }) => (React.createElement(S.Wrapper, null,
    React.createElement(S.Title, null, "Your personalized incentives"),
    React.createElement(S.IncentivesBody, null,
        React.createElement(S.IncentivesBodyIcon, null,
            React.createElement(Icon, { icon: 'Lightning' }),
            React.createElement(Icon, { icon: 'Lightning' }),
            React.createElement(Icon, { icon: 'Lightning' })),
        React.createElement("div", null,
            React.createElement(S.DescriptionWrapper, null,
                React.createElement(S.DescriptionContainer, null,
                    React.createElement(S.Subtitle, null,
                        "Available tax credits",
                        React.createElement(Info, { text: 'The geothermal tax credit is not included in this total.' })),
                    React.createElement(S.Price, { "data-testid": "household_tax_savings" }, formatCurrency(householdInformation === null || householdInformation === void 0 ? void 0 : householdInformation.tax_savings))),
                React.createElement(S.DescriptionContainer, null,
                    React.createElement(S.Subtitle, null,
                        "Estimated energy savings per year",
                        React.createElement(Info, { text: `This estimate is based on energy costs in your state. 
                It includes savings on both utility and gasoline bills if you switch to an 
                electric vehicle and a heat pump for space and water heating.` })),
                    React.createElement(S.Price, { "data-testid": "household_annual_savings" }, formatCurrency(householdInformation === null || householdInformation === void 0 ? void 0 : householdInformation.estimated_annual_savings)))),
            React.createElement(S.Total, null, "Total Incentives"),
            React.createElement(S.TotalPrice, { "data-testid": "household_total_savings" }, formatCurrency((householdInformation === null || householdInformation === void 0 ? void 0 : householdInformation.tax_savings) +
                (householdInformation === null || householdInformation === void 0 ? void 0 : householdInformation.pos_savings))))),
    React.createElement(S.Separator, null),
    React.createElement(S.IncentivesFooter, null,
        React.createElement(S.IncentivesFooterContent, null, "Disclaimer: These values are estimates. The rebates may be implemented differently in each state, so we cannot guarantee final amounts, eligibility, or timeline. And without additional appropriations from Congress, the rebate programs will end once their initial IRA funding is exhausted. Tax credits can only be used to offset your federal taxes owed, which we estimate but do not know."),
        React.createElement(S.IncentivesFooterContent, null, "Your income is above the threshold for upfront discounts and/or EV tax credits, but your tax liability will likely qualify you for equipment tax credits. You do qualify for a performance-based efficiency rebate worth $4,000. However, we do not consider this an upfront discount. Check out this relevant case study!"))));
export default PersonalizedIncentives;
