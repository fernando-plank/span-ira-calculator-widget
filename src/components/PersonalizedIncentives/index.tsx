import { formatCurrency } from '@utils/currency'

import * as S from './styles'
import Info from '@components/Info'
import Icon from '@components/Icon'
import { IncentivesProps } from '../../pages/Home'
import { MutableRefObject } from 'react'

export interface PersonalizedIncentivesProps {
  householdInformation: IncentivesProps
  incentivesRef?: MutableRefObject<any>
}

const PersonalizedIncentives = ({
  householdInformation,
  incentivesRef
}: PersonalizedIncentivesProps) => (
  <S.Wrapper>
    <div
      style={{ height: '65px', backgroundColor: '#FAFAFA' }}
      ref={incentivesRef}
    ></div>
    <S.Background>
      <S.Content>
        <S.Title>Your personalized incentives</S.Title>
        <S.IncentivesBody>
          <S.IncentivesBodyIcon>
            <Icon icon={'Lightning'} />
            <Icon icon={'Lightning'} />
            <Icon icon={'Lightning'} />
          </S.IncentivesBodyIcon>
          <div style={{ flex: '0 0 50%' }}>
            <S.DescriptionWrapper>
              <S.DescriptionBox>
                <S.DescriptionContainer>
                  <S.Subtitle>Available</S.Subtitle>
                  <S.Subtitle>
                    tax credits
                    <Info
                      text={
                        'The geothermal tax credit is not included in this total.'
                      }
                    />
                  </S.Subtitle>
                  <S.Price data-testid="household_tax_savings">
                    {formatCurrency(householdInformation?.tax_savings)}
                  </S.Price>
                </S.DescriptionContainer>
                <S.DescriptionContainer>
                  <S.Subtitle>Estimated energy</S.Subtitle>
                  <S.Subtitle>
                    savings per year
                    <Info
                      text={`This estimate is based on energy costs in your state. 
                It includes savings on both utility and gasoline bills if you switch to an 
                electric vehicle and a heat pump for space and water heating.`}
                    />
                  </S.Subtitle>
                  <S.Price data-testid="household_annual_savings">
                    {formatCurrency(
                      householdInformation?.estimated_annual_savings
                    )}
                  </S.Price>
                </S.DescriptionContainer>
              </S.DescriptionBox>
              <S.TotalContainer>
                <S.TotalText>Total incentives</S.TotalText>
                <S.TotalPrice>
                  {formatCurrency(
                    householdInformation?.estimated_annual_savings
                  )}
                </S.TotalPrice>
              </S.TotalContainer>
            </S.DescriptionWrapper>
          </div>
        </S.IncentivesBody>
        <S.Separator></S.Separator>
        <S.IncentivesFooter>
          <S.IncentivesFooterContent>
            <p>
              These values are estimates. Tax credits can only be used to offset
              your federal taxes owed (our estimate may be inaccurate).
            </p>
            <p style={{ marginTop: '15px' }}>
              Powered by the{' '}
              <S.ContentLink
                href="https://www.rewiringamerica.org"
                target="_blank"
              >
                Rewiring America API
              </S.ContentLink>{' '}
              and subject to the API{' '}
              <S.ContentLink
                href="https://content.rewiringamerica.org/api/terms.pdf"
                target="_blank"
              >
                Terms of Service.
              </S.ContentLink>
            </p>
          </S.IncentivesFooterContent>
          <S.IncentivesFooterContent className="hide-on-mobile">
            Your income is above the threshold for upfront discounts and/or EV
            tax credits, but your tax liability will likely qualify you for
            equipment tax credits. You do qualify for a performance-based
            efficiency rebate worth $4,000. However, we do not consider this an
            upfront discount. Check out this relevant case study!
          </S.IncentivesFooterContent>
        </S.IncentivesFooter>
      </S.Content>
    </S.Background>
  </S.Wrapper>
)

export default PersonalizedIncentives
