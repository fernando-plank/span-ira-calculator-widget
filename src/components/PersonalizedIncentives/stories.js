import React from 'react';
import PersonalizedIncentives from '.';
export default {
    title: 'PersonalizedIncentives',
    component: PersonalizedIncentives
};
export const Default = () => (React.createElement(PersonalizedIncentives, { householdInformation: {
        estimated_annual_savings: 0,
        tax_savings: 0,
        pos_savings: 0
    } }));
