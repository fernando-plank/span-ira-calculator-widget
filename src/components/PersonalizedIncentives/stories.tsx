import React from 'react'
import { Story, Meta } from '@storybook/react'

import PersonalizedIncentives from '.'

export default {
  title: 'PersonalizedIncentives',
  component: PersonalizedIncentives
} as Meta

export const Default: Story = () => (
  <PersonalizedIncentives
    householdInformation={{
      estimated_annual_savings: 0,
      tax_savings: 0,
      pos_savings: 0
    }}
    incentivesRef={undefined}
  />
)
