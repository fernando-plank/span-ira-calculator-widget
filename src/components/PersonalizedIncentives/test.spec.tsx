import { screen } from '@testing-library/react'
import { render } from '@utils/test-utils'

import PersonalizedIncentives from '.'

const personalizedIncentivesProps = {
  pos_savings: 123,
  tax_savings: 320,
  estimated_annual_savings: 111
}

jest.mock('../Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

describe('<PersonalizedIncentives />', () => {
  it('should render the tax savings', () => {
    const { container } = render(
      <PersonalizedIncentives
        householdInformation={personalizedIncentivesProps}
        incentivesRef={undefined}
      />
    )

    expect(screen.getByTestId('household_tax_savings')).toHaveTextContent(
      '$' + personalizedIncentivesProps.tax_savings
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the household anual savings', () => {
    const { container } = render(
      <PersonalizedIncentives
        householdInformation={personalizedIncentivesProps}
        incentivesRef={undefined}
      />
    )

    expect(screen.getByTestId('household_annual_savings')).toHaveTextContent(
      '$' + personalizedIncentivesProps.estimated_annual_savings
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
