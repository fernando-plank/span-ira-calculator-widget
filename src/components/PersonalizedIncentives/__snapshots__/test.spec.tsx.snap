// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalizedIncentives /> should render the household anual savings 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  background-color: #787BFF;
}

.c2 svg {
  margin: 0px 10px;
}

.c3 {
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  text-align: left;
  margin-bottom: 42px;
  color: #FAFAFA;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 50px;
}

.c5 {
  width: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  height: 30vh;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  color: #FAFAFA;
}

.c12 {
  margin-top: 44px;
  color: #FAFAFA;
}

.c13 {
  font-size: 48px;
  line-height: 56px;
  font-weight: 700;
}

.c14 {
  font-size: 48px;
  line-height: 56px;
  font-weight: 700;
}

.c9 {
  font-style: normal;
  font-weight: 500;
  font-size: 24px;
  line-height: 24px;
  text-transform: uppercase;
  text-align: left;
  max-width: 100vw;
}

.c9 svg > path {
  fill: white;
}

.c9 svg > circle {
  stroke: white;
}

.c11 {
  font-size: 38px;
  font-weight: 700;
  line-height: 40px;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  text-align: left;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 40px;
}

.c17 {
  color: #FAFAFA !important;
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
  text-align: left;
  margin-bottom: 30px;
}

.c18 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
  font-size: 12px;
  margin: 12px 0px 9px 0px;
  font-weight: bold;
}

.c15 {
  border: 1px solid #FAFAFA;
}

.c10 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: contents;
}

@media (min-width:1024px) {
  .c0 {
    padding: 0px 32px;
  }
}

@media (min-width:1528px) {
  .c0 {
    padding: 40px 0px;
  }
}

@media (max-width:425px) {
  .c0 .hide-on-mobile {
    display: none;
  }
}

@media (min-width:320px) {
  .c1 {
    padding: 50px 24px;
  }
}

@media (min-width:480px) {
  .c1 {
    padding: 50px 32px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 50px 40px;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding: 40px 32px;
  }
}

@media (min-width:1528px) {
  .c2 {
    padding: 0px;
    width: 1400px;
    margin: 0px auto;
  }
}

@media (min-width:1024px) {
  .c3 {
    width: 30vw;
  }
}

@media (min-width:320px) {
  .c4 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media (min-width:1024px) {
  .c4 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (max-width:425px) {
  .c5 {
    position: absolute;
    margin-top: 150px;
  }
}

@media (max-width:768px) {
  .c5 {
    height: auto;
  }
}

@media (min-width:1024px) {
  .c6 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c7 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c8 {
    margin-right: 50px;
  }
}

@media (min-width:320px) {
  .c8 {
    margin-top: 28px;
  }
}

@media (min-width:1024px) {
  .c14 {
    font-size: 136px;
    font-weight: 600;
    line-height: 136px;
  }
}

@media (max-width:1024px) {
  .c16 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c17 {
    padding-right: 59px;
    -webkit-flex: 0 0 50%;
    -ms-flex: 0 0 50%;
    flex: 0 0 50%;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h1
        class="c3"
      >
        Your personalized incentives
      </h1>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div>
            mocked icon
          </div>
          <div>
            mocked icon
          </div>
          <div>
            mocked icon
          </div>
        </div>
        <div
          style="margin-bottom: 60px; flex: 0 0 50%;"
        >
          <div
            class="c6"
          >
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <h6
                  class="c9"
                >
                  Available
                </h6>
                <h6
                  class="c9"
                >
                  tax credits
                  <div
                    class="c10"
                    data-testid="tooltip"
                  >
                    <div>
                      mocked icon
                    </div>
                  </div>
                </h6>
                <h6
                  class="c11"
                  data-testid="household_tax_savings"
                >
                  $320
                </h6>
              </div>
              <div
                class="c8"
              >
                <h6
                  class="c9"
                >
                  Estimated energy
                </h6>
                <h6
                  class="c9"
                >
                  savings per year
                  <div
                    class="c10"
                    data-testid="tooltip"
                  >
                    <div>
                      mocked icon
                    </div>
                  </div>
                </h6>
                <h6
                  class="c11"
                  data-testid="household_annual_savings"
                >
                  $111
                </h6>
              </div>
            </div>
            <div
              class="c12"
            >
              <h1
                class="c13"
              >
                Total incentives
              </h1>
              <h1
                class="c14"
              >
                $111
              </h1>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c15"
      />
      <div
        class="c16"
      >
        <p
          class="c17"
        >
          <p>
            These values are estimates. Tax credits can only be used to offset your federal taxes owed (our estimate may be inaccurate).
          </p>
          <p
            style="margin-top: 15px;"
          >
            Powered by the
             
            <a
              class="c18"
              href="https://www.rewiringamerica.org"
              target="_blank"
            >
              Rewiring America API
            </a>
             
            and subject to the API
             
            <a
              class="c18"
              href="https://content.rewiringamerica.org/api/terms.pdf"
              target="_blank"
            >
              Terms of Service.
            </a>
          </p>
        </p>
        <p
          class="c17 hide-on-mobile"
        >
          Your income is above the threshold for upfront discounts and/or EV tax credits, but your tax liability will likely qualify you for equipment tax credits. You do qualify for a performance-based efficiency rebate worth $4,000. However, we do not consider this an upfront discount. Check out this relevant case study!
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PersonalizedIncentives /> should render the tax savings 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  background-color: #787BFF;
}

.c2 svg {
  margin: 0px 10px;
}

.c3 {
  font-style: normal;
  font-weight: 700;
  font-size: 32px;
  line-height: 40px;
  text-align: left;
  margin-bottom: 42px;
  color: #FAFAFA;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 50px;
}

.c5 {
  width: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  height: 30vh;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  color: #FAFAFA;
}

.c12 {
  margin-top: 44px;
  color: #FAFAFA;
}

.c13 {
  font-size: 48px;
  line-height: 56px;
  font-weight: 700;
}

.c14 {
  font-size: 48px;
  line-height: 56px;
  font-weight: 700;
}

.c9 {
  font-style: normal;
  font-weight: 500;
  font-size: 24px;
  line-height: 24px;
  text-transform: uppercase;
  text-align: left;
  max-width: 100vw;
}

.c9 svg > path {
  fill: white;
}

.c9 svg > circle {
  stroke: white;
}

.c11 {
  font-size: 38px;
  font-weight: 700;
  line-height: 40px;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  text-align: left;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 40px;
}

.c17 {
  color: #FAFAFA !important;
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
  text-align: left;
  margin-bottom: 30px;
}

.c18 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
  font-size: 12px;
  margin: 12px 0px 9px 0px;
  font-weight: bold;
}

.c15 {
  border: 1px solid #FAFAFA;
}

.c10 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: contents;
}

@media (min-width:1024px) {
  .c0 {
    padding: 0px 32px;
  }
}

@media (min-width:1528px) {
  .c0 {
    padding: 40px 0px;
  }
}

@media (max-width:425px) {
  .c0 .hide-on-mobile {
    display: none;
  }
}

@media (min-width:320px) {
  .c1 {
    padding: 50px 24px;
  }
}

@media (min-width:480px) {
  .c1 {
    padding: 50px 32px;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 50px 40px;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding: 40px 32px;
  }
}

@media (min-width:1528px) {
  .c2 {
    padding: 0px;
    width: 1400px;
    margin: 0px auto;
  }
}

@media (min-width:1024px) {
  .c3 {
    width: 30vw;
  }
}

@media (min-width:320px) {
  .c4 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media (min-width:1024px) {
  .c4 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (max-width:425px) {
  .c5 {
    position: absolute;
    margin-top: 150px;
  }
}

@media (max-width:768px) {
  .c5 {
    height: auto;
  }
}

@media (min-width:1024px) {
  .c6 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c7 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c8 {
    margin-right: 50px;
  }
}

@media (min-width:320px) {
  .c8 {
    margin-top: 28px;
  }
}

@media (min-width:1024px) {
  .c14 {
    font-size: 136px;
    font-weight: 600;
    line-height: 136px;
  }
}

@media (max-width:1024px) {
  .c16 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1024px) {
  .c17 {
    padding-right: 59px;
    -webkit-flex: 0 0 50%;
    -ms-flex: 0 0 50%;
    flex: 0 0 50%;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h1
        class="c3"
      >
        Your personalized incentives
      </h1>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div>
            mocked icon
          </div>
          <div>
            mocked icon
          </div>
          <div>
            mocked icon
          </div>
        </div>
        <div
          style="margin-bottom: 60px; flex: 0 0 50%;"
        >
          <div
            class="c6"
          >
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <h6
                  class="c9"
                >
                  Available
                </h6>
                <h6
                  class="c9"
                >
                  tax credits
                  <div
                    class="c10"
                    data-testid="tooltip"
                  >
                    <div>
                      mocked icon
                    </div>
                  </div>
                </h6>
                <h6
                  class="c11"
                  data-testid="household_tax_savings"
                >
                  $320
                </h6>
              </div>
              <div
                class="c8"
              >
                <h6
                  class="c9"
                >
                  Estimated energy
                </h6>
                <h6
                  class="c9"
                >
                  savings per year
                  <div
                    class="c10"
                    data-testid="tooltip"
                  >
                    <div>
                      mocked icon
                    </div>
                  </div>
                </h6>
                <h6
                  class="c11"
                  data-testid="household_annual_savings"
                >
                  $111
                </h6>
              </div>
            </div>
            <div
              class="c12"
            >
              <h1
                class="c13"
              >
                Total incentives
              </h1>
              <h1
                class="c14"
              >
                $111
              </h1>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c15"
      />
      <div
        class="c16"
      >
        <p
          class="c17"
        >
          <p>
            These values are estimates. Tax credits can only be used to offset your federal taxes owed (our estimate may be inaccurate).
          </p>
          <p
            style="margin-top: 15px;"
          >
            Powered by the
             
            <a
              class="c18"
              href="https://www.rewiringamerica.org"
              target="_blank"
            >
              Rewiring America API
            </a>
             
            and subject to the API
             
            <a
              class="c18"
              href="https://content.rewiringamerica.org/api/terms.pdf"
              target="_blank"
            >
              Terms of Service.
            </a>
          </p>
        </p>
        <p
          class="c17 hide-on-mobile"
        >
          Your income is above the threshold for upfront discounts and/or EV tax credits, but your tax liability will likely qualify you for equipment tax credits. You do qualify for a performance-based efficiency rebate worth $4,000. However, we do not consider this an upfront discount. Check out this relevant case study!
        </p>
      </div>
    </div>
  </div>
</div>
`;
